awscognito/main_test.go

```diff
@@ -136,6 +136,8 @@ func TestSendCognitoEvent(t *testing.T) {
 	httpmock.Activate()
 	defer httpmock.DeactivateAndReset()
 
+	identityPoolID = "fooPool"
```

Comment:
This value is read [from an environment variable](https://github.com/triggermesh/knative-lambda-sources/blob/5605f3a4da7a061da437a7aa7b7112500e49e36e/awscognito/main.go#L72) inside the main. I think the correct thing to do here is to initialize the env vars you need [inside your test](https://github.com/kyma-project/kyma/blob/master/components/event-sources/reconciler/httpsource/controller_test.go#L55-L56) to remain close to what the code you test actually does.

Reply:
the issue is `main.go` picks up the environment values (global variables) in the `init` call, setting `os.Setenv` in the test would get called after `os.GetEnv`, so it's too late. 

I could possibly fetch the env variables in the calls instead of having global variables. Do you think it's the right thing to do or is there some alternate method?

https://github.com/triggermesh/knative-lambda-sources/blob/master/awscognito/main.go#L68-L76

Reply:
Right, I did not pay attention to the `init`.

A good practice would be to avoid `init` and package globals altogether, but that's not the topic of this PR, so I guess that approach is fine for the current code ✅